Issue #120: Updated ldavis.js to be compatible with latest d3.v5

dtemkin opened the pull request:
CHANGED 
  scale.linear(...)
TO
  scaleLinear(...)

CHANGED
    .axis(...).scale(...).orient(...) 
TO
    axisLeft (_y_)(...), axisTop(...) (_x_) or axisBottom(...) (_scaleX_) 

CHANGED
    d3.scaleordinal(...).domain(...).rangeRoundBands(...)
TO
    d3.scaleBand().domain(...).rangeBands(...).padding(...)

CHANGED
    y.rangeBand(...) 
TO
    y.bandwidth(...)

REPLACED d3.v3.js with d3.v5.js

ADDED:
    optional function parameters that let user specify primary (color1) and 
secondary (color2) colors. (Default colors are originals)



This should also be compatible with d3.v4 but I havent tested it yet.

dtemkin commented:
I looked at the errors and they appear to be the result of an issue in pandas. I can attempt to address them if necessary to merge my PR.

bmabey commented:
Could you resolve the conflicts in ldavis.js?

msusol commented:
Are these changes still relevant? I can make an attempt to resolve the conflicts in ldavis.js

Is there any other effort to upgrade d3.v3 to d3.v5+ ?

dtemkin commented:
@msusol @bmabey Sorry this took me so long to complete. I fixed the references to d3 so that they are specifically referencing d3.v5.min.js or https://d3js.org/d3.v5.js . I believe everything is good to go, but take a look when you get a chance to make sure I am not missing something. 

I also fixed the problems with the travis-ci configuration in which I had to separate the dependencies for python 3.5.

bmabey commented on pyLDAvis/_prepare.py:
```diff
@@ -188,12 +188,11 @@ def _series_with_name(data, name):
         return pd.Series(data, name=name)
 
 
-def _topic_coordinates(mds, topic_term_dists, topic_proportion, start_index=1):
```
Is there a reason why you changed this line?

bmabey commented on pyLDAvis/_prepare.py:
```diff
@@ -188,12 +188,11 @@ def _series_with_name(data, name):
     mds_res = mds(topic_term_dists)
     assert mds_res.shape == (K, 2)
-    mds_df = pd.DataFrame({'x': mds_res[:, 0], 'y': mds_res[:, 1],
-                           'topics': range(start_index, K + start_index),
```
Ditto here.. my guess is that this is an accident...

bmabey commented on pyLDAvis/_prepare.py:
```diff
@@ -217,24 +216,22 @@ def _job_chunks(l, n_jobs):
 def _find_relevance(log_ttd, log_lift, R, lambda_):
     relevance = lambda_ * log_ttd + (1 - lambda_) * log_lift
-    return relevance.T.apply(lambda topic: topic.nlargest(R).index)
+    return relevance.T.apply(lambda s: s.sort_values(ascending=False).index).head(R)
```
ditto

bmabey commented on pyLDAvis/_prepare.py:
```diff
@@ -217,24 +216,22 @@ def _job_chunks(l, n_jobs):
 
 
 def _topic_info(topic_term_dists, topic_proportion, term_frequency, term_topic_freq,
-                vocab, lambda_step, R, n_jobs, start_index=1):
```
ditto, I don't see why these changes are needed for this PR

bmabey reviewed (changes requested):
It looks like you maybe forgot to rebase to get the latest changes? Python code changes are happening that seem unrelated to updating to the latest d3.

bmabey commented:
LGTM, thank you for this @dtemkin!

msusol commented:
Thank you!!

msusol commented:
Thank you, Dan!

Ben, when will there be a new version tag available?

On Wed, Dec 2, 2020 at 11:10 AM Ben Mabey <notifications@github.com> wrote:

> Merged #154 <https://github.com/bmabey/pyLDAvis/pull/154> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bmabey/pyLDAvis/pull/154#event-4063725653>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAVPINZMP4ORNTGV7MMGVBLSSZ7JLANCNFSM4IPCPCVQ>
> .
>

msusol commented:
Dan,

Please see the recent issues about missing the labels in the d3.v5
rendering.

On Thu, Dec 17, 2020, 9:12 AM Mark Susol <marksusol@gmail.com> wrote:

> Thank you, Dan!
>
> Ben, when will there be a new version tag available?
>
> On Wed, Dec 2, 2020 at 11:10 AM Ben Mabey <notifications@github.com>
> wrote:
>
>> Merged #154 <https://github.com/bmabey/pyLDAvis/pull/154> into master.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/bmabey/pyLDAvis/pull/154#event-4063725653>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAVPINZMP4ORNTGV7MMGVBLSSZ7JLANCNFSM4IPCPCVQ>
>> .
>>
>

msusol commented:
See follow up refactoring: #198
